homeassistant/components/stib_mivb/sensor.py

```diff
@@ -0,0 +1,275 @@
+"""Support for STIB-MIVB (Brussels public transport) information."""
+import datetime
+import logging
+import math
+
+from pyodstibmivb import HttpException, ODStibMivb
+import pytz
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, DEVICE_CLASS_TIMESTAMP
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Data provided by opendata.stib-mivb.be"
+
+CONF_STOPS = "stops"
+CONF_STOP_ID = "stop_id"
+CONF_API_KEY = "api_key"
+CONF_LANG = "lang"
+CONF_MESSAGE_LANG = "message_lang"
+CONF_LINE_NUMBERS = "line_numbers"
+DEFAULT_NAME = "Stib-Mivb"
+
+SUPPORTED_LANGUAGES = ["nl", "fr"]
+SUPPORTED_MESSAGE_LANGUAGES = ["en", "nl", "fr"]
+
+TYPES = {
+    "0": "tram",
+    "1": "subway",
+    "3": "bus",
+}
+
+STOP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Required(CONF_LINE_NUMBERS): [cv.string],
+    }
+)
+
+STOPS_SCHEMA = vol.All(cv.ensure_list, [STOP_SCHEMA])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
I'll look into it. Looks really complicated compared to yaml config though...